src/systems/rendering.cpp

```diff
@@ -11,6 +11,21 @@
 
 std::vector<RenderSystem::RenderHandler> RenderSystem::render_handlers;
 
+bool MeshRenderComponent::ensureLoaded()
```

Comment:
This would probably be less error prone if the fetching/caching logic were moved into `getters` for the fields in question.  It's annoying to have to make sure to call `ensureLoaded()` in the right place before using mesh fields.